src/pages/ListSwaps/LedgerActionDialogBody.tsx

```diff
@@ -11,14 +11,19 @@ import Web3SendTransactionButton from "../../components/Web3SendTransactionButto
 import CopyToClipboardButton from "./CopyToClipboard";
 
 interface LedgerActionDialogBodyProps {
-  onClose: (event: React.MouseEvent) => void;
+  onClose: () => void;
   action: LedgerAction;
 }
 
 function LedgerActionDialogBody({
   onClose,
   action
 }: LedgerActionDialogBodyProps) {
+  function onSuccess() {
+    onClose();
+    location.reload();
```

Comment:
Triggering a reload in an SPA is a bit dodgy. Why did you have to add that?

Reply:
I didn't have to, but otherwise after performing an ethereum action the button is still there, because the resource (either the swap list or a particular swap) hasn't been fetched again.

Reply:
Okay, @da-kami and me are going to revise these dialogs now anyway so it's fine.